### PR TITLE
Fix unselectable code host status messages

### LINKED ISSUE #14643: Impossible to copy errors out of code host status indicator
When repository synching errors occur, they appear in this UI but it is **impossible** to select the text or copy and paste it:

image

The same is true on k8s.sgdev.org without errors right now. I also confirmed this was an issue at least as early as Sourcegraph v3.19

@sourcegraph/cloud are you or @sourcegraph/web the right owner for this?

### PR DESCRIPTION
**Pull request by artemruts**
Tested in Dev environment with invalid code host connection:  

test

@felixfbecker Do you want me figure out a better way in SASS?

Fixes #14643

### CODE REVIEW

**Comment by felixfbecker** [comment]
The default for `user-select` is to allow selection, so some ancestor element must be overriding it. I'm curious which element that is? Should it maybe be changed there?

**Comment by artemruts** [comment]
> The default for `user-select` is to allow selection, so some ancestor element must be overriding it. I'm curious which element that is? Should it maybe be changed there?

https://sourcegraph.com/github.com/sourcegraph/sourcegraph@45f9839011ac5df56d1cb07dfa09c7749dd26084/-/blob/client/web/src/nav/NavLinks.scss?utm_source=chrome-extension#L6

I think it makes sense to keep navs not "user-select"able.

**Comment by felixfbecker** [comment]
Ah, I see. I'd override the whole dropdown with `user-select: contain` then. Wdyt?

**Comment by artemruts** [comment]
> Ah, I see. I'd override the whole dropdown with `user-select: contain` then. Wdyt?

Felix, maybe let's explicitly ignore links, so every `status-messages-nav-item__entry` is selectable if it's not a `_link`?

**Comment by felixfbecker** [comment]
I think it's generally fine to have links still be selectable, because that's standard behavior on the web. Something like the title of the dropdown should also still be selectable, not just the items. Basically imo everything should be selectable by default unless we have a good reason why they shouldn't be. Only buttons should not be selectable, which navlinks are more like, or e.g. links in the file tree which also are styled more like buttons.

**Comment by tsenart** [comment]
@artemruts: Can you give a more descriptive title to the PR please? Makes it easier to navigate around our team's GitHub issues and PRs.

**Comment by artemruts** [comment]
> I think it's generally fine to have links still be selectable, because that's standard behavior on the web. Something like the title of the dropdown should also still be selectable, not just the items. Basically imo everything should be selectable by default unless we have a good reason why they shouldn't be. Only buttons should not be selectable, which navlinks are more like, or e.g. links in the file tree which also are styled more like buttons.

That makes sense, useful information 🙇 

I think `user-select: contain` is not supported https://caniuse.com/mdn-css_properties_user-select_contain, definitely didn't work in chrome / ff.

Let's just allow for all nav dropdown menus?

**Comment by felixfbecker** [comment]
Yeah we can just use `user-select: auto` then. Thanks for checking!

**Comment by artemruts** [comment]
> Yeah we can just use `user-select: auto` then. Thanks for checking!

I don’t think `auto` will work because `auto` will become `none`,  
since  `nav-links`  > `nav-link`  > `dropdown` and all of our `nav-* ` have explicit `user-select: none`


If `dropdown` classes have `auto` they’ll use parent’s `none` (as per [MDN](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select#Syntax))

**Comment by felixfbecker** [comment]
Right, what you have seems right then! (sorry I only gave the docs a brief look)

File: client/web/src/nav/StatusMessagesNavItem.scss [hunk 1]
```diff
@@ -7,6 +7,7 @@
         min-width: 22rem;
         max-width: 22rem;
         padding: 1rem 1rem 0 1rem;
+        user-select: text;
```
Comment (felixfbecker) [inline client/web/src/nav/StatusMessagesNavItem.scss:10]: Any reason to not allow `all`? (feel free to merge straight away regardless of the answer)
```suggestion
        user-select: all;
```

**Review by felixfbecker — APPROVED** [review]

File: client/web/src/nav/StatusMessagesNavItem.scss (hunk 1 above)
Comment (artemruts) [reply client/web/src/nav/StatusMessagesNavItem.scss:10]: I think, selection is a bit more natural. I'll merge it but we can always revisit. Thank you, Felix!

**Pull request merged**